src/app/group/group-add-edit/group-add-edit.component.ts

```diff
@@ -11,68 +12,180 @@ import { filter } from 'rxjs';
       }
     })
-  }
 
-  // Add member name and member email formgroup in add group form
-  memberNameAndEmailForm(): FormGroup{
+    if (this.getActivatedRouteParam != undefined) {
+      this.addGroupForm.patchValue({
+        groupName: 'trip',
+        about: 'this is group for testing',
+        groupMember: [{
+          userId: 'admin'
+        }]
+      })
+    }
+
+
+    /** On click of group edit button it will set current details of group to textbox */
+    if (this.getActivatedRouteParam != undefined) {
+      const data = this.groupsService.groupList[Number(this.getActivatedRouteParam)];
+      this.addGroupForm.patchValue({
+        groupName: data.name,
+        about: data.about,
+        groupMember: [{
+          userId: data.createdBy
+        }]
+      })
+    }
+
+
+
+  }// OnInit method end
+
+  /** Add member name and member email formgroup in add group form
+   * @returns FormGroup
+   **/
+  addMembersToGroupForm(): FormGroup {
     return new FormGroup({
-      memberName : new FormControl(''),
-      memberEmail : new FormControl('')
+      memberName: new FormControl(''),
+      memberEmail: new FormControl('')
     })
   }
 
-  // Getter methods
-  get groupName(){
-    return this.addGroupForm.get('groupName');
+  /** Get groupName FormControl
+   *  @returns FormControl returns groupName
+   **/
+  get groupName(): FormControl {
+    return this.addGroupForm.get('groupName') as FormControl;
   }
 
-  // getter of dyanamic create group member
-  get memberNameEmail(){
-    return this.addGroupForm.get('groupMember') as FormArray;
+  /** Get groupDescription FormControl
+   *  @returns FormControl returns about
+   **/
+  get about(): FormControl {
+    return this.addGroupForm.get('about') as FormControl;
   }
 
-  // Add member dynamically
-  addMemberNameAndEmail(): void{
-    this.memberNameEmail.push(this.memberNameAndEmailForm());
+  /** Get groupMember FromArray
+   *  @returns FromArray returns groupMembers
+   **/
+  get groupMembers(): FormArray {
+    return this.addMembersForm.get('groupMembers') as FormArray;
   }
 
-  // Delete member name and email
-  deleteMemberNameAndEmail(i : number): void{
-    this.memberNameEmail.removeAt(i);
+  /** Add member field in form
+   *  on click of add menber button
+   **/
+  addMemberNameAndEmailField(): void {
+    this.groupMembers.push(this.addMembersToGroupForm());
+  }
+
+  /** Delete member field in form
+   *  on click of Delete button
+   *  @param index index of field which is selected to delete
+   **/
+  deleteMemberNameAndEmailField(i: number): void {
+    this.groupMembers.removeAt(i);
+  }
+
+  /** create group method to get data from add group form
+   * and send to post api call function
+   **/
+  createGroup() {
+    const data: Groups = {
+      name: this.groupName?.value as string,
+      about: this.about?.value as string,
+      groupMembers: [{
+        userId: "3a0ba79a-0f40-0dd1-4e91-9ed501777180"
+      }]
+    }
```

Comment:
Semi colon

Reply:
It showing error when i try to add semi colon here.